src/apdu_sign.c

```diff
@@ -459,9 +476,11 @@ void finish_input_cell_data() {
             G.maybe_transaction.v.tag = OPERATION_TAG_DAO_PREPARE;
         }
     } else {
-        if(G.cell_state.data_size !=0) {
+        if(G.cell_state.data_size != 0 && N_data.contract_data_type == DISALLOW_CONTRACT_DATA) {
             REJECT("Data found in non-dao cell");
```

Comment:
Should we reword this?

Reply:
There may be a better way to say it, but I think it's still accurate - we allow data in the DAO cell by default, but not anywhere else.

Reply:
I've added more details on the new setting